source/fundamentals/connection/mongoclientsettings.txt

```diff
@@ -0,0 +1,367 @@
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ConnectionPoolSettings
+   :end-before: end ConnectionPoolSettings
+   :language: java
+   :emphasize-lines: 3-5
+   :dedent:
+
+.. _mcs-server-settings:
+
+Server Settings
+---------------
+
+Chain the `applyToServerSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToServerSettings(com.mongodb.Block)>`__
+method to modify the driver's behavior when monitoring each MongoDB
+server.
+
+The following table describes all the methods you can chain to your
+settings to modify the driver's behavior:
+
+.. list-table::
+   :widths: 40 60
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``addServerListener()``
+     - Adds a listener for server-related events.
+
+   * - ``addServerMonitorListener​()``
+     - Adds a listener for server monitor-related events.
+
+   * - ``applyConnectionString​()``
+     - Uses the settings from a ``ConnectionString`` object.
+
+   * - ``applySettings()``
+     - Uses the server settings specified in a ``ServerSettings`` object.
+
+   * - ``heartbeatFrequency()``
+     - Sets the interval for a cluster monitor to attempt reaching a server.
+
+   * - ``minHeartbeatFrequency()``
+     - Sets the interval for server monitoring checks.
+
+Example
+~~~~~~~
+
+This example specifies the following driver behavior in a MongoDB server:
+
+- To interval between server monitoring checks to be at least ``700 MILLISECONDS`` 
+- The cluster monitor to attempt reaching a server every ``15 SECONDS``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ServerSettings
+   :end-before: end ServerSettings
+   :language: java
+   :emphasize-lines: 3-5
+   :dedent:
+
+.. _mcs-ssl-settings:
+
+TLS/SSL Settings
+----------------
+
+Chain the `applyToSslSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToSslSettings(com.mongodb.Block)>`__
+method to modify the driver's behavior when using TLS/SSL to secure a
+connection between your application and MongoDB.
+
+The following table describes all the methods you can chain to your
+settings to modify the driver's behavior:
+
+.. list-table::
+   :widths: 40 60
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``applyConnectionString​()``
+     - Uses the settings from a ``ConnectionString`` object.
+
+   * - ``applySettings()``
+     - Uses the TLS/SSL settings specified in a ``SslSettings`` object.
+
+   * - ``context()``
+     -  Sets the secure socket protocol implementation to use when you enable TLS/SSL.
+
+   * - ``enabled()``
+     - Whether to ensure your Atlas connection string works.
+
+   * - ``invalidHostNameAllowed()``
+     - Whether to allow a mismatch between the server’s hostname and the hostname specified by the TLS certificate.
+
+Example
+~~~~~~~
+
+This example specifies the driver to ensure your Atlas connection string
```

Comment:
"specifies the driver" seems ungrammatical.  Can you rework this sentence (and perhaps similar ones above)?

Also, I don't think we should make this Atlas-specific.  Ops Manager also enables TLS, for example, as do many customers who deploy MongoDB themselves.